components/ListServices/details.jsx

```diff
@@ -1,39 +1,85 @@
+import { useCallback } from 'react';
 import { useRouter } from 'next/router';
-import { Result } from 'antd';
 
 import Details from 'common-util/Details';
 import { useHelpers } from 'common-util/hooks';
+import { useSvmConnectivity } from 'common-util/hooks/useSvmConnectivity';
+
 import {
   getServiceDetails,
-  getServiceHashes,
   getServiceOwner,
-  getTokenUri,
+  getTokenUri as getEvmTokenUri,
 } from './utils';
+import {
+  useGetServiceDetails,
+  useServiceOwner,
+  useTokenUri,
+} from './useSvmService';
+import { ServiceState } from './ServiceState';
+import { EmptyMessage } from '../styles';
 
 const Service = () => {
   const router = useRouter();
   const id = router?.query?.id;
   const { links, isSvm } = useHelpers();
+  const { hasNoSvmPublicKey } = useSvmConnectivity();
+  const { getSvmServiceDetails } = useGetServiceDetails();
+  const { getSvmServiceOwner } = useServiceOwner();
+  const { getSvmTokenUri } = useTokenUri();
+
+  const getDetails = useCallback(async () => {
+    const e = isSvm
+      ? await getSvmServiceDetails(id)
+      : await getServiceDetails(id);
+    return e;
+  }, [id, isSvm, getSvmServiceDetails]);
+
+  const getOwner = useCallback(async () => {
+    const e = isSvm ? await getSvmServiceOwner(id) : await getServiceOwner(id);
+    return e;
+  }, [id, isSvm, getSvmServiceOwner]);
+
+  const getTokenUri = useCallback(async () => {
+    const e = isSvm ? await getSvmTokenUri(id) : await getEvmTokenUri(id);
+    return e;
+  }, [id, isSvm, getSvmTokenUri]);
+
+  const handleUpdate = useCallback(() => {
+    router.push(`${links.UPDATE_SERVICE}/${id}`);
+  }, [id, router, links]);
+
+  const navigateToDependency = useCallback(
+    (e) => {
+      router.push(`${links.AGENTS}/${e}`);
+    },
+    [router, links],
+  );
 
-  if (isSvm) {
+  if (hasNoSvmPublicKey) {
     return (
-      <Result
-        status="warning"
-        title="This page is not available for SVM services yet!"
-      />
+      <EmptyMessage>
+        Please connect a wallet that hold SOL to view this page.
```

Comment:
```suggestion
        To view this page, connect a wallet that holds SOL
```

In general, best to lead this kind of message with the action the user is trying to take, and follow up with the action they need to take to resolve the issue.